File: frontend/lib/phone.ts
const LIBYA_MOBILE_REGEX = /^\+2189[1-6]\d{7}$/;

export interface NormalizedPhone {
  e164: string;
  digits: string;
  local: string;
}

export function normalizeLibyaPhone(raw: string): NormalizedPhone {
  let phone = raw.replace(/[\s\-\(\)\.]+/g, "").trim();

  if (phone.startsWith("00218")) {
    phone = "+" + phone.slice(2);
  } else if (phone.startsWith("00")) {
    phone = "+" + phone.slice(2);
  } else if (phone.startsWith("218")) {
    phone = "+" + phone;
  } else if (phone.startsWith("0")) {
    phone = "+218" + phone.slice(1);
  }

  if (!LIBYA_MOBILE_REGEX.test(phone)) {
    throw new Error("رقم الهاتف غير صحيح. يرجى إدخال رقم ليبي صحيح.");
  }

  const digits = phone.slice(1);
  const local = "0" + digits.slice(3);
  return { e164: phone, digits, local };
}

export function isValidLibyaPhone(raw: string): boolean {
  try {
    normalizeLibyaPhone(raw);
    return true;
  } catch {
    return false;
  }
}

File: backend/app/services/phone.py
import re
from dataclasses import dataclass


@dataclass(frozen=True)
class LibyaPhone:
    e164: str
    digits: str
    local: str


def normalize_libya_phone(raw: str) -> LibyaPhone | None:
    value = re.sub(r"[\s\-().]", "", raw.strip())
    value = re.sub(r"[^\d+]", "", value)

    if value.startswith("00"):
        value = f"+{value[2:]}"
    elif value.startswith("218"):
        value = f"+{value}"
    elif value.startswith("0"):
        value = f"+218{value[1:]}"

    if not re.fullmatch(r"\+2189[1-6]\d{7}", value):
        return None

    return LibyaPhone(e164=value, digits=value.replace("+", ""), local=f"0{value[4:]}")
